packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1438,9 +1441,36 @@ class _NestedScrollPosition extends ScrollPosition implements ScrollActivityDele
     coordinator.updateCanDrag();
   }
 
-  void updateCanDrag(double totalExtent) {
-    context.setCanDrag(physics.allowUserScrolling &&
-        (totalExtent > (viewportDimension - maxScrollExtent) || minScrollExtent != maxScrollExtent));
+  void updateCanDrag({
+    required double totalInnerExtent,
+    required double overlapExtent
+  }) {
+    // This is only called for the outer position.
+    // Any overlap extent is already deducted from the SliverGeometry of any
+    // SliverOverlapAbsorber wrapped slivers, so the maxExtent of the outer
+    // position does not reflect the extent that has been absorbed. This is as
+    // expected since the inner and outer scroll views' extents are manipulated
+    // in this way for them to behave as one, but the extent must be accounted
+    // for when considering the full extent available to drag.
+    // For example, a pinned SliverAppBar in the outer position with a minExtent
+    // and maxExtent of 56.0 will have a ScrollPosition.maxExtent of 0.0 after
+    // the SliverOverlap is applied. This is correct because the pinned sliver
+    // app bar should not have to consume scrolling before the inner body of the
+    // NestedScrollView scrolls. However, when calculating the drag here, 0
+    // could result in scrolling being disabled unexpectedly if the inner scroll
+    // view by itself were not long enough to allow scrolling on its own.
```

Comment:
Great comment!
I have to admit that this was pretty tricky for me to wrap my head around this ;-)

I am not very familiar with the internals of NestedScrollView, so I'll leave this to @xu-baolin to review

Reply:
I was actually able to make it even simpler. :)